src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/Checkout/UpdateCart.xml

```diff
@@ -23,6 +23,7 @@
             <group>sylius:shop:cart:update</group>
         </attribute>
         <attribute name="email">
+            <group>sylius:shop:cart:update</group>
```

Comment:
As a logged-in user, I can now change my email. However, there is another group `sylius:shop:guest:write` added by the  `ShopGuestGroupsContextBuilder` when a guest makes a request, allowing the guest to change the email, but not the logged-in user.

Reply:
No, you can't. I added this field to the serialization to inform the client about the API's behavior.
We have a validator responsible for handling this:
https://github.com/Sylius/Sylius/pull/16760/files#diff-f4b2a01a4abdb0ea87e702a4c865f77b7dea2f4ebe8e9143d5e6715619e4b291R51-R52
but until now it was never enabled:
https://github.com/Sylius/Sylius/pull/16760/files#diff-9407ff0efc53fadf40739eaaab45a10e09b1159553f98d6c72ef163c647ee901R253-R260

I believe it's a good idea to inform the client of the outcome of their actions, in this case, why they can't change a specific field. What do you think?

Reply:
We chose to resign from ContextBuilder in favor of validation.